euphoria-hadoop/src/main/java/cz/seznam/euphoria/hadoop/output/SequenceFileSink.java

```diff
@@ -29,21 +33,81 @@
  */
 public class SequenceFileSink<K, V> extends HadoopSink<K, V> {
 
-  // TODO provide factory: how to parse out the keyType and valueType info from the URI?
-
   /**
-   * Convenience constructor delegating to
-   * {@link #SequenceFileSink(Class, Class, String, Configuration)} with a newly
-   * created hadoop configuration.
    *
-   * @param keyType the type of the key consumed and emitted to the output
-   *         by the new data sink
-   * @param valueType the type of the value consumed and emitted to the output
-   *         by the new data sink
-   * @param path the destination where to place the output to
+   * @param keyType the class representing the type of the keys emitted
+   * @param valueType the class representing the type of the values emitted
+   * @param <K> type of the keys emitted
+   * @param <V> type of the values emitted
+   * @return Builder
    */
-  public SequenceFileSink(Class<K> keyType, Class<V> valueType, String path) {
-    this(keyType, valueType, path, new Configuration());
+  public static <K, V> Builder<K, V> newBuilder(Class<K> keyType, Class<V> valueType) {
```

Comment:
This looks imbalanced, if we use `Builder`, I'd suggest to create the builder as empty and set all parameters via methods, e.g. `withKey` and `withValue`, or `withKeyValue`. This way these two parameters have some "special" handling. This would be understandable if it was the case, that *all* required arguments would be set in `newBuilder` method, but output path isn't and therefore it would be better to handle all parameters equally.

Reply:
I put mandatory fields in newBuilder as param, because otherwise you will need to set explicit type for key and value via `SequenceFileSink.<Text,Text>newBuilder()..` and thats very ugly.

Reply:
I think that is not entirely true, because the `newBuilder` can return untyped `Builder` object, which can then gather type by additional call to e.g. `withKeyValue` and return correctly typed `Builder` object. You can have an inspiration about how to do this in builders for operators - i.e. `FlatMap`.